rest-services/server/src/main/java/com/_4point/aem/docservices/rest_services/server/assembler/AssembleDocuments.java

```diff
@@ -0,0 +1,209 @@
+			String concatenatedPDFName = (String) docsMap.getKey();
+			if (concatenatedPDFName.equalsIgnoreCase("concatenatedPDF.pdf")) {
+				Object pdf = docsMap.getValue();
+				concatenatedDoc = (Document) pdf;
+				// concatenatedPDF =
+				// org.apache.commons.io.IOUtils.toByteArray(concatenatedDoc.getInputStream());
+				break;
+			}
+		}
+		return concatenatedDoc;
+	}
+
+	private InputStream createDdxFile(SlingHttpServletRequest request, Map<String, Object> mapOfDocuments)
+			throws Exception {
+		InputStream xmlInputStream = null;
+
+		final boolean isMultipart = org.apache.commons.fileupload.servlet.ServletFileUpload.isMultipartContent(request);
+		DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
+		DocumentBuilder docBuilder = null;
+		try {
+			docBuilder = docBuilderFactory.newDocumentBuilder();
+			org.w3c.dom.Document ddx = docBuilder.newDocument();
+			Element mainRootElement = ddx.createElementNS("http://ns.adobe.com/DDX/1.0/", "DDX");
+			ddx.appendChild(mainRootElement);
+			mainRootElement.appendChild(getPDFNodes(ddx, isMultipart, mapOfDocuments, request));
+
+			DOMSource domSource = new DOMSource(ddx);
+			ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+			Result outputTarget = new StreamResult(outputStream);
+
+			TransformerFactory.newInstance().newTransformer().transform(domSource, outputTarget);
+			xmlInputStream = new ByteArrayInputStream(outputStream.toByteArray());
+
+		} catch (Exception e) {
+			throw new Exception("Error creating ddx xml file", e);
+		}
+
+		return xmlInputStream;
+	}
+
+	private Node getPDFNodes(org.w3c.dom.Document ddx, boolean isMultipart, Map<String, Object> mapOfDocuments,
+			SlingHttpServletRequest request) {
+		Element pdfResult = ddx.createElement("PDF");
+		pdfResult.setAttribute("result", "concatenatedPDF.pdf");
+		if (isMultipart) {
+			Map<String, org.apache.sling.api.request.RequestParameter[]> params = request.getRequestParameterMap();
+			for (Map.Entry<String, org.apache.sling.api.request.RequestParameter[]> pairs : params.entrySet()) {
+				final RequestParameter[] pArr = pairs.getValue();
+				final RequestParameter param = pArr[0];
+
+				try {
+					if (!param.isFormField()) {
+						final InputStream stream = param.getInputStream();
+						log.debug("the file name is " + param.getFileName());
+						log.debug("Got input Stream inside my servlet####" + stream.available());
+						Document document = docFactory.create(stream);
+						mapOfDocuments.put(param.getFileName(), document);
+						Element pdfSourceElement = ddx.createElement("PDF");
+						pdfSourceElement.setAttribute("source", param.getFileName() + ".pdf");
+						pdfResult.appendChild(pdfSourceElement);
+						log.debug("The map size is " + mapOfDocuments.size());
+					} else {
+						log.debug("The form field is" + param.getString());
+
+					}
+				} catch (IOException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
```

Comment:
Again, leaving the autogenerated code in a generated try/batch block is almost never correct.  If an IOException occurs, then wrap it in a custom exception type and re-throw the exception.  Have a look at the [Output service's `GeneratePdfOutput` class](https://github.com/4PointSolutions/FluentFormsAPI/blob/master/rest-services/server/src/main/java/com/_4point/aem/docservices/rest_services/server/output/GeneratePdfOutput.java).  All checked exceptions are wrapped in either a BadRequestException or an InternalServerErrorException so that we can return back the correct HTTP Status code (and so that the caller knows whether the problem is theirs or ours).